client/src/index.ts

```diff
@@ -262,14 +362,16 @@ export abstract class Repository<T> extends Query<T, Selectable<T>> {
 
   abstract create(object: Selectable<T>): Promise<T>;
 
+  abstract createMany(objects: Selectable<T>[]): Promise<T[]>;
+
   abstract read(id: string): Promise<T | null>;
 
   abstract update(id: string, object: Partial<T>): Promise<T>;
 
   abstract delete(id: string): void;
 
   // Used by the Query object internally
-  abstract query<R>(query: Query<T, R>): Promise<R[]>;
+  abstract executeQuery<R>(query: Query<T, R>, options?: BulkQueryOptions<T>): Promise<Page<T, R>>;
```

Comment:
I like the name change 👍 

Do you think we should add an underscore as a prefix?